util/types/field_type.go

```diff
@@ -62,6 +62,69 @@ func NewFieldType(tp byte) *FieldType {
 	}
 }
 
+// AggFieldType aggregates field types.
+func AggFieldType(tps []*FieldType) *FieldType {
+	var currType FieldType
+	for _, t := range tps {
+		if currType.Tp == mysql.TypeUnspecified {
+			currType = *t
+			continue
+		}
+		mtp := MergeFieldType(currType.Tp, t.Tp)
+		currType.Tp = mtp
+	}
+	return &currType
+}
+
+// AggTypeClass aggregates type classes.
+func AggTypeClass(tps []*FieldType, flag *uint) TypeClass {
+	var (
+		tpClass      = ClassString
+		unsigned     bool
+		gotFirst     bool
+		gotBinString bool
+	)
+	for _, argFieldType := range tps {
+		if argFieldType.Tp == mysql.TypeNull {
+			continue
+		}
+		argTypeClass := argFieldType.ToClass()
+		if argTypeClass == ClassString && mysql.HasBinaryFlag(argFieldType.Flag) {
+			gotBinString = true
+		}
+		if !gotFirst {
+			gotFirst = true
+			tpClass = argTypeClass
+			unsigned = mysql.HasUnsignedFlag(argFieldType.Flag)
+		} else {
+			tpClass = mergeTypeClass(tpClass, argTypeClass, unsigned, mysql.HasUnsignedFlag(argFieldType.Flag))
+			unsigned = unsigned && mysql.HasUnsignedFlag(argFieldType.Flag)
+		}
+	}
+	setTypeFlag(flag, uint(mysql.UnsignedFlag), unsigned)
+	setTypeFlag(flag, uint(mysql.BinaryFlag), tpClass != ClassString || gotBinString)
+	return tpClass
+}
+
+func mergeTypeClass(a, b TypeClass, aUnsigned, bUnsigned bool) TypeClass {
```

Comment:
Is this the same work with https://github.com/pingcap/tidb/blob/master/util/types/field_type.go#L306 ？